cpp/job/tests.cpp

```diff
@@ -3,85 +3,108 @@
 #include "job_manager.h"
 #include "../log/log.h"
 
+#include <thread>
+#include <chrono>
+
 namespace openage {
 namespace job {
 namespace tests {
 
 
 int test_simple_job() {
-	JobManager manager{4};
-	manager.start();
-
-	bool finished = false;
-	bool result = -1;
-
-	auto job_function = []() -> int {
-		return 42;
-	};
-	auto job_callback = [&](result_function_t<int> get_result) {
-		int job_result = get_result();
-		if (job_result == 42) {
-			result = 0;
-		}
-		finished = true;
-
-	};
-	manager.enqueue<int>(job_function, job_callback);
-
-	while (not finished) {
-		manager.execute_callbacks();
-	}
-
-	manager.stop();
-	return result;
+    JobManager manager{4};
+    manager.start();
+
+    std::atomic<int> finish_count(0);
+    int job_count = 10;
+    bool result = -1;
+
+    auto job_function = []() -> int {
+        std::this_thread::sleep_for(std::chrono::seconds(1));
```

Comment:
sleeping that long is probably a bad idea, we're running the test on our ci server, and waiting for long times doesn't speed up its run.